libraries/AP_Scripting/applets/copter_terrain_brake.lua

```diff
@@ -0,0 +1,130 @@
+--[[
+script to prevent terrain impact in LOITER mode while flying copters in steep terrain
+--]]
+
+local PARAM_TABLE_KEY = 84
+local PARAM_TABLE_PREFIX = "TERR_BRK_"
+
+local MODE_LOITER = 5
+local MODE_BRAKE = 17
+
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+   assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+   return Parameter(PARAM_TABLE_PREFIX .. name)
+end
+
+-- setup script specific parameters
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 14), 'could not add param table')
+
+--[[
+  // @Param: TERR_BRK_ENABLE
+  // @DisplayName: terrain brake enable
+  // @Description: terrain brake enable
+  // @Values: 0:Disabled,1:Enabled
+  // @User: Standard
+--]]
+local TERR_BRK_ENABLE = bind_add_param('ENABLE', 1, 1)
+
+--[[
+  // @Param: TERR_BRK_ALT
+  // @DisplayName: terrain brake altitude
+  // @Description: terrain brake altitude. The altitude above the ground below which BRAKE mode will be engaged if in LOITER mode.
+  // @Range: 1 100
+  // @Units: m
+  // @User: Standard
+--]]
+local TERR_BRK_ALT = bind_add_param('ALT', 2, 30)
+
+--[[
+  // @Param: TERR_BRK_HDIST
+  // @DisplayName: terrain brake home distance
+  // @Description: terrain brake home distance. The distance from home where the auto BRAKE will be enabled. When within this distance of home the script will not activate
+  // @Range: 0 1000
+  // @Units: m
+  // @User: Standard
+--]]
+local TERR_BRK_HDIST = bind_add_param('HDIST', 3, 100)
+
+--[[
+  // @Param: TERR_BRK_SPD
+  // @DisplayName: terrain brake speed threshold
+  // @Description: terrain brake speed threshold. Don't trigger BRAKE if both horizontal speed and descent rate are below this threshold. By setting this to a small value this can be used to allow the user to climb up to a safe altitude in LOITER mode. A value of 0.5 is recommended if you want to use LOITER to recover from an emergency terrai BRAKE mode change.
```

Comment:
```suggestion
  // @Description: terrain brake speed threshold. Don't trigger BRAKE if both horizontal speed and descent rate are below this threshold. By setting this to a small value this can be used to allow the user to climb up to a safe altitude in LOITER mode. A value of 0.5 is recommended if you want to use LOITER to recover from an emergency terrain BRAKE mode change.
```

Reply:
fixed